crates/walrus-service/src/node.rs

```diff
@@ -2928,29 +3183,78 @@ impl ServiceState for StorageNodeInner {
         blob_id: BlobId,
         sliver_pair_index: SliverPairIndex,
         sliver: Sliver,
+        intent: UploadIntent,
     ) -> Result<bool, StoreSliverError> {
         self.check_index(sliver_pair_index)?;
+        let (metadata, persisted) = self
```

Comment:
nit: can we be a bit more explicit here and maybe call the variable `metadata_persisted`